metaconfig-tests/shared/src/test/scala/metaconfig/ConfListSuite.scala

```diff
@@ -0,0 +1,119 @@
+package metaconfig
+
+object ConfListSuite {
+
+  case class Bar(add: List[String] = Nil)
+
+  implicit val barSurface: generic.Surface[Bar] =
+    generic.deriveSurface
+
+  implicit val barReader: ConfDecoder[Bar] =
+    generic.deriveDecoder[Bar](Bar()).noTypos
+
+  case class Foo(
+      field: List[String] = List("c"),
+      anotherField: List[String] = Nil,
+      bar: Bar = Bar()
+  )
+  implicit val surface: generic.Surface[Foo] =
+    generic.deriveSurface
+
+  implicit val reader: ConfDecoder[Foo] =
+    generic.deriveDecoder[Foo](Foo()).noTypos
+
+  case class Nested(raw: String = "x")
```

Comment:
there are multi-nested fields in `AllTheAnnotations`, can we test deeper?

Reply:
Added test with multiple nesting